File: book-service/src/main/java/qm/service/book/model/Book.java
package qm.service.book.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.math.BigDecimal;
import java.util.Objects;

/**
 * Book -
 */
@Entity
@Table(name = "books")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    private String author;

    @Column(unique = true, length = 20)
    private String isbn;

    @Column(precision = 10, scale = 2)
    private BigDecimal price = BigDecimal.ZERO;

    @Override
    public final boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Book that)) return false;
        return getId() != null && Objects.equals(getId(), that.getId());
    }

    @Override
    public final int hashCode() {
        return Book.class.hashCode();
    }
}

File: library-service/src/main/java/qm/service/library/model/Book.kt
package qm.service.library.model

import io.quarkus.hibernate.reactive.panache.kotlin.PanacheCompanion
import io.quarkus.hibernate.reactive.panache.kotlin.PanacheEntityBase
import jakarta.persistence.*
import java.math.BigDecimal

/**
 * Book -
 */
@Entity
@Table(name = "books")
class Book : PanacheEntityBase {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @Column(nullable = false)
    lateinit var title: String
    var author: String? = null

    @Column(unique = true, length = 20)
    var isbn: String? = null

    @Column(precision = 10, scale = 2)
    var price: BigDecimal? = null

    companion object : PanacheCompanion<Book>

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Book) return false
        return id != null && id == other.id
    }

    override fun hashCode(): Int {
        return Book::class.java.hashCode()
    }
}